ament_flake8/ament_flake8/main.py

```diff
@@ -192,32 +205,31 @@ def get_xunit_content(report, testname):
     return xml
 
 
-class CustomStyleGuide(flake8.engine.NoQAStyleGuide):
+class CustomReport(object):
+
+    def __init__(self):
+        self.files = []
+        self.errors = []
+        self.report = None
+
+    def add_filename(self, filename):
+        self.files.append(filename)
 
-    def input_file(self, filename, **kwargs):
-        self.options.reporter.files.append(filename)
-        return super(CustomStyleGuide, self).input_file(filename, **kwargs)
+    @property
+    def total_errors(self):
+        return len(self.errors)
 
+    def add_error(self, error):
+        self.errors.append(error)
 
-class CustomReport(pep8.StandardReport):
+    def get_error_count(self):
+        return len([e for e in self.errors if e.code.startswith('E')])
 
-    errors = []
-    files = []
+    def get_warning_count(self):
+        return len([e for e in self.errors if not e.code.startswith('E')])
```

Comment:
I am curious for the motivation behind distinguishing the different codes, because I think we seem to treat them all as bad as each other when reviewing code. [pep257 error codes](http://pep257.readthedocs.io/en/latest/error_codes.html) will start with D and [import order error codes](https://github.com/PyCQA/flake8-import-order/blob/c63331f422dc552963b5a0dbd34192caf25c9371/README.rst) will start with I, [flake8 with F](https://flake8.readthedocs.io/en/2.0/warnings.html), and these will be counted as warnings